src/graph/connect.cc

```diff
@@ -631,7 +631,8 @@ ncclResult_t ncclTopoPostset(struct ncclComm* comm, int* firstRanks, int* treePa
   // Define channels for gfx94 GPU architectures
   if (IsArchMatch(comm->topo->nodes[GPU].nodes[0].gpu.gcn, "gfx94")) {
     // Only use full MAXCHANNELS for gfx94x
-    maxChannels = MAXCHANNELS;
+    // Use max 80 channels in case of 80 CUs
+    maxChannels = comm->topo->nodes[GPU].nodes[0].gpu.cu == 80 ? 80 : MAXCHANNELS;
```

Comment:
Is 80 special, or should this be something like
```suggestion
    maxChannels = min(comm->topo->nodes[GPU].nodes[0].gpu.cu, MAXCHANNELS);
```

Reply:
since this PR reuses "sm" field, it contains some random numbers like "90", "94", "98"... in older XMLs.

Reply:
alternatively, we need to fix all existing xmls with CU count